### PR TITLE
Process messages in same order as they are received

### PR DESCRIPTION
**Pull request by bhardwajaditya**

### CODE REVIEW

File: lib/payloadAction.ts
Comment (Shailesh351) [inline lib/payloadAction.ts:86]: Can be shortened to `isFallback,`

File: lib/payloadAction.ts
Comment (Shailesh351) [inline lib/payloadAction.ts:85]: Calling `createDialogflowMessage` in `handlePayloadActions` function is not a good idea. Only job of `handlePayloadActions` should be to handle payload action.

If required we should create, separate fn to `handleResponse`, which will then call `handlePayloadActions` and `createDialogFlowMessage` functions based on response type. (If action call  `handlePayloadActions` else `createDialogFlowMessage`)

**Review by Shailesh351 — COMMENTED** [review]
### 🛑 **While testing this PR. customer idle timeout was broken as a side effect.** 

### Steps
1) Start a chat with the bot
2) Send close chat action with multiple text responses from Dialogflow
3) Chat is ended but the timer is still running

Screenshot 2021-08-28 at 11 24 03 AM

Screenshot 2021-08-28 at 10 54 24 AM

When I switch back to master, the issue is not there

**Comment by bhardwajaditya** [comment]
@Shailesh351 the issue you mentioned regarding the timer, it's there with master code as well. I think it's there in Livechat Widget rather than Dialogflow app. I will make a separate story for that and fix it in Livechat widget

**Comment by ear-dev** [comment]
@bhardwajaditya can you look at the conflicting file please?

**Comment by bhardwajaditya** [comment]
@ear-dev @chadgoss fixed this PR

**Comment by ear-dev** [comment]
@bhardwajaditya please resolve the conflicts.

**Comment by ear-dev** [comment]
@bhardwajaditya I tested this and it looks good.......  but fwiw....  I am not seeing the timer issue that @Shailesh351 mentioned. 

I'm checking with @chadgoss about https://github.com/WideChat/Rocket.Chat/issues/901.  I think we will have to wait until that is ready before merge.

**Comment by chadgoss** [comment]
@bhardwajaditya I'm seeing that multiple custom payloads are not working,

for example: this should disable input, display the texts, then escalate. It escalates, but I never see the 


Screen Shot 2021-09-16 at 2 20 36 PM

**Comment by bhardwajaditya** [comment]
@chadgoss we currently don't handle disable input custom payloads from DF, that's why it's not working

**Comment by ear-dev** [comment]
@bhardwajaditya what is the current mechanism by which we disable input?

**Comment by bhardwajaditya** [comment]
We do that based on message type from DF and SF apps. Like with welcome message we have added it in DF app and before welcome intent. For Salesforce we do that with Queue messages, but these disable input actions are not handled.

**Comment by Shailesh351** [comment]
@bhardwajaditya In this file https://github.com/WideChat/Apps.Dialogflow/blob/master/lib/Message.ts#L13 It sends message along with customFields.

**Comment by bhardwajaditya** [comment]
@Shailesh351 thanks for this, it never worked for me so I thought we don't handle this. 

@chadgoss I will look into this bug.

**Comment by ear-dev** [comment]
@bhardwajaditya can you resolve these conflicts please, and add some comments about where we are with this PR... thanks!

**Comment by bhardwajaditya** [comment]
So for this one, it's not how dialogflow send us , it depends on how we process it. Currently we assign the custom fields part of response payload, to the last message of the response, be it an action or text whichever is the last message. That's why the custom field of disable input does not maintain the order intended for it.

@ear-dev @chadgoss Do we want to change it and treat it like a standalone message with a custom field in it?

**Comment by ear-dev** [comment]
@chadgoss I think that is what we want?

**Comment by chadgoss** [comment]
> @chadgoss I think that is what we want?

that sounds correct, yep

**Comment by bhardwajaditya** [comment]
@ear-dev @chadgoss fixed it

**Review by chadgoss — APPROVED** [review]

**Pull request merged**
